### PR TITLE
Inbox

### PR DESCRIPTION
**Pull request by TheHollidayInn**
Add inbox support to the app. 

my Habitica User-ID:

### CODE REVIEW

File: Habitica/res/values/strings.xml [hunk 1]
```diff
@@ -347,4 +347,7 @@ To start, which parts of your life do you want to improve?</string>
     <string name="open_in_store">Open in play store</string>
     <string name="change_class_confirmation">Are you sure you want to change your class? This will cost 3 gems.</string>
     <string name="leaderMessage" formatted="false">Message from %1$s</string>
+
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (vIiRuS) [inline Habitica/res/values/strings.xml]: this isn't needed, right?

File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/menu/MainDrawerBuilder.java [hunk 2]
```diff
@@ -32,6 +33,7 @@
     // Change the identificationIDs to the position IDs so that its easier to set the selected entry
     public static final int SIDEBAR_TASKS = 0;
     public static final int SIDEBAR_SKILLS = 1;
+    public static final int SIDEBAR_INBOX = 15;
```
Comment (vIiRuS) [inline Habitica/src/main/java/com/habitrpg/android/habitica/ui/menu/MainDrawerBuilder.java]: can you change the IDs to be corresponding to their position? (so inbox is 3, tavern 4, etc.)

**Comment by vIiRuS** [comment]
Currently inbox messages are never marked as read. Could you add, so that they are marked as read when the user opens the inbox fragment? We currently don't have a way to have individual messages marked as unread, so marking all as read when the overview fragment is openened is the only way.

Having an indicator that there are unread messages would also be good. The materialdrawer library should have an option to either add such an indicator or create a custom drawer item with an indicator. The indicator should either just be a dot or a a circle with the number of unread messages.

**Comment by TheHollidayInn** [comment]
Updated!

**Pull request merged**